src/workloads/docs/CrudActorFSM.yml

```diff
@@ -0,0 +1,157 @@
+        - WriteCommand: insertOne
+          Document: {phoneID: 3, currentPhoneState: "PhoneIsOff"}
+        - WriteCommand: insertOne
+          Document: {phoneID: 4, currentPhoneState: "PhoneIsOff"}
+        - WriteCommand: insertOne
+          Document: {phoneID: 5, currentPhoneState: "PhoneIsOff"}
+    - &nop {Nop: true}
+    - *nop
+
+- Name: StateSmartPhoneDemo
+  Type: Crud
+  Threads: 5
+  Database: StateTest
+  Phases:
+  - *nop
+  - Duration: 30 seconds
+    Collection: CrudDemo
+    States:
+    - Name: "On"  # The device is on
+
+      # The states support the keywords Repeat, SleepBefore, SleepAfter, (local) Rate, and
+      # Duration. These keywords behave the same as in Phase Control, but operate per state (i.e.,
+      # per actor instance as it goes through the state machine.
+      Repeat: 1
+
+      # A state can have (optional) a set of operations that execute upon entering the state. This
+      # state updates a specific document to indicate it is in state On.
+
+      # Note: Following the analogy to phases -- phases may be blocking or non-blocking. States are
+      # always non-blocking, in the sense that any sleepBefore, SleepAfter delays will be
+      # interrupted to close the phase. However, any active operations will be finished.
+
+      # For example, this phase has a 30 second duration, making it blocking. When that 30 seconds
+      # end all the actors will end their phases. If an actor is in a sleepBefore or sleepAfter
+      # delay, it will exit the phase. If the actor is executing a state's operations, it will
+      # complete those operations, and then exit the phase.
+      Operations:
+      - OperationName: updateOne
+        OperationCommand:
+          Filter: &filter {phoneID: {^ActorId: {}}}
+          Update:
+            $set: {currentPhoneState: "PhoneIsOn"}
+
+      # Transitions specifies next states, delays, and operations. The On state can transition to
+      # each of the three other states
+      Transitions:
+      - To: "Off"  # Transition to this state
+
+        # Transitions are probabilistically chosen based on weight. A transition with weight 2 is
+        # twice as likely to be picked as a transition with weight 1. Weights must be greater than
+        # 0.
+        Weight: 1
+
+        # Transitions also support the phase keywords of sleepBefore, sleepAfter, and repeat.  For
+        # sleepBefore, the system is considered to remain in it's current state. If there is a
+        # repeating action, it will continue repeating until the transition fires.  For repeat, the
+        # operation is repeated immediately.  sleepAfter will not be supported for transitions with
+        # no operations.
+
+        # ^TimeSpec is a proposed generator to randomly generate acceptable TimeSpecs with units. It
+        # accepts any valid int or double generator or constant for value.
+        SleepBefore: {^TimeSpec: {value: {^RandomDouble: {min: 0, max: 15}}, unit: "seconds"}}
```

Comment:
Same comment / suggested example text above - the `Transition`s always respect the blocking behavior of the Phase they're in, ya?

Reply:
Added a comment in the comment block above.